pydra/engine/specs.py

```diff
@@ -39,13 +39,13 @@ def collect_additional_outputs(self, input_spec, inputs, output_dir):
     @property
     def hash(self):
         """Compute a basic hash for any given set of fields."""
-        from .helpers import hash_function
+        from .helpers import hash_function, ensure_list
         from .helpers_file import hash_file
 
         inp_dict = {
             field.name: hash_file(getattr(self, field.name))
             if (field.type == File and "container_path" not in field.metadata)
-            else getattr(self, field.name)
+            else ensure_list(getattr(self, field.name), tuple2list=True)
```

Comment:
this is turning every input into a list.

Reply:
i've changed it